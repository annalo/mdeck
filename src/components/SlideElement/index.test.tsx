import React, { useContext } from "react";
import { render, screen } from "@testing-library/react";

import {
  SlideshowObserver,
  SlideshowObserverProvider,
} from "contexts/SlideshowObserver";
import { SlideElement } from ".";

const component = (
  <SlideElement attributes={{}} elementTag="h1" srcLine={2}>
    <div>Some Child Element</div>
  </SlideElement>
);
describe("<SlideElement />", () => {
  test("should render and match the snapshot", () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the element type from elementTag prop", () => {
    render(component);
    const header = screen.getByRole("heading");
    expect(header).toBeInTheDocument();
  });

  test("rendered element should have data-line attribute", () => {
    const { container } = render(component);
    expect(container.firstChild).toHaveAttribute("data-line");
  });

  // TODO not sure what's a better way to test that the value of
  // the context was updated without testing an actual dom element
  test("observes itself in SlideshowObserver after render", () => {
    const TestObserverEntriesComponent = () => {
      const { entries } = useContext(SlideshowObserver);
      return <div data-testid="test-entries-length">{entries.length}</div>;
    };

    render(
      <SlideshowObserverProvider>
        {component}
        <TestObserverEntriesComponent />
      </SlideshowObserverProvider>
    );

    expect(screen.getByTestId("test-entries-length")).toHaveTextContent("1");
  });
});
