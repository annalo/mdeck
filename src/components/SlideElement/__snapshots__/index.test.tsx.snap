// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideElement /> should render and match the snapshot 1`] = `
<DocumentFragment>
  <h1
    data-line="2"
  >
    <div>
      Some Child Element
    </div>
  </h1>
</DocumentFragment>
`;
